### PR TITLE
feat(ecs): add Amazon ECS cypress tests

### PR DESCRIPTION
**Pull request by pkandasamy91**
Add tests to test the ECS pipelines view and cluster/server groups view.

Screen Shot 2020-09-24 at 11 56 59 PM

### CODE REVIEW

**Review by caseyhebebrand — APPROVED** [review]
LGTM

**Review by McNouvion — APPROVED** [review]

File: app/scripts/modules/ecs/src/ecsCluster/ecsCluster.read.service.ts [hunk 1]
```diff
@@ -3,7 +3,7 @@ import { module } from 'angular';
 import { API } from '@spinnaker/core';
 import { IEcsClusterDescriptor } from './IEcsCluster';
 
-export class EscClusterReader {
+export class EcsClusterReader {
```
Comment (allisaurus) [inline app/scripts/modules/ecs/src/ecsCluster/ecsCluster.read.service.ts:6]: bah! Nice catch 😎

File: test/functional/cypress/integration/ecs/pipelines.spec.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+    cy.get('a:contains("Deploy")').click({ force: true });
+
+    cy.get('.glyphicon-duplicate')
+      .eq(1)
+      .click({ force: true });
+
+    cy.get('.account-tag').should('have.length', 2);
+
+    cy.get('.account-tag')
```
Comment (allisaurus) [inline test/functional/cypress/integration/ecs/pipelines.spec.js:59]: looks like we're checking whether both/all values contain the same text. Would it make sense/be possible to loop over them vs. individually inspecting them?

**Review by allisaurus — COMMENTED** [review]
So so happy to see these come together! 🎉 😭 

I'm not really familiar with cypress, so added a couple questions about how stuff is fitting together plus a couple nits/observations which may not be impactful.

File: test/functional/cypress/fixtures/ecs/pipelines/pipelineConfigs.json [hunk 3]
```diff
@@ -0,0 +1,151 @@
+[
+  {
+    "expectedArtifacts": [
```
[the remark this answers is not in the document]
Comment (pkandasamy91) [reply test/functional/cypress/fixtures/ecs/pipelines/pipelineConfigs.json:3]: This mimics real output that I'm seeing in my existing pipeline config.

File: test/functional/cypress/fixtures/ecs/clusters/serverGroup.ecsdemo-v000.json [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+        },
+        {
+          "healthClass": "platform",
+          "instanceId": "630cbec0-afa2-4529-80dd-5d84c9f7f4fa",
+          "state": "Up",
+          "type": "ecs"
+        }
+      ],
+      "healthState": "Up",
+      "humanReadableName": "630cbec0-afa2-4529-80dd-5d84c9f7f4fa",
```
[the remark this answers is not in the document]
Comment (pkandasamy91) [reply test/functional/cypress/fixtures/ecs/clusters/serverGroup.ecsdemo-v000.json:41]: Sadly, this mimics real output

File: test/functional/cypress/fixtures/default/networks.aws.json [hunk 5]
```diff
@@ -0,0 +1,17 @@
+[
+  {
+    "account": "my-aws-devel-acct",
```
[the remark this answers is not in the document]
Comment (pkandasamy91) [reply test/functional/cypress/fixtures/default/networks.aws.json:3]: This mimics real output that I'm seeing in my existing network configuration.

File: test/functional/cypress/fixtures/default/applications.json [hunk 6]
```diff
@@ -12,6 +12,19 @@
     "updateTs": "1544636963981",
     "user": "[anonymous]"
   },
+  {
+    "cloudProviders": "ecs",
+    "createTs": "1519160914577",
+    "dataSources": { "disabled": [], "enabled": ["canaryConfigs", "serverGroups", "loadBalancers", "securityGroups"] },
```
[the remark this answers is not in the document]
Comment (pkandasamy91) [reply test/functional/cypress/fixtures/default/applications.json:18]: I haven't tried to scope down which props are needed (I used real output that I'm seeing in network calls that I intercepted)

File: test/functional/cypress/integration/ecs/pipelines.spec.js (hunk 2 above)
Comment (pkandasamy91) [reply test/functional/cypress/integration/ecs/pipelines.spec.js:59]: Possibly, I haven't found a nice way of doing this just yet :(

File: test/functional/cypress/fixtures/ecs/pipelines/pipelineConfigs.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (allisaurus) [reply test/functional/cypress/fixtures/ecs/pipelines/pipelineConfigs.json:3]: Sure, if your pipeline is using an expected artifact from the pipeline, which is the "legacy" way of doing things. I'm wondering if there's anything in here though that won't work if we define the artifact inline? We don't necessarily have to write this test yet but I want to have a plan to test the current/preferred way of defining an artifact.

[the remark this answers is not in the document]
Comment (pkandasamy91) [reply test/functional/cypress/fixtures/ecs/pipelines/pipelineConfigs.json:3]: There's a plan to test creating a new server group with artifacts. In this test, I wanted to make sure that the existing configurations worked (considering many customers might not be using the new way of defining artifacts)

[the remark this answers is not in the document]
Comment (allisaurus) [reply test/functional/cypress/fixtures/ecs/pipelines/pipelineConfigs.json:3]: ok, so just to recap where I think this landed:
1. looks like the `expectedArtifacts` part of this pipeline config is not actually asserted on in the test which loads it, so I'm guessing changing it here would not make a difference?
2. the plan is to add a test which creates a server group with an inline (new/now standard) artifact, so we will be exercising this code path in the functional tests.

If these are correct than I'm on board.

[the remark this answers is not in the document]
Comment (pkandasamy91) [reply test/functional/cypress/fixtures/ecs/pipelines/pipelineConfigs.json:3]: Exactly!

**Review by allisaurus — APPROVED** [review]

**Pull request merged**